README.md

```diff
@@ -1,2 +1,216 @@
+
+#### Adding Service Principal permissions
+
+Your service principal will need a few permissions added to it, for these you'll need to get its `objectId`, you can get this by running:
+
+```bash
+$ az ad sp list --display-name "<name-for-your-service-principal>"
+[
+  {
+    "accountEnabled": "True",
+    "addIns": [],
+    "alternativeNames": [],
+    "appDisplayName": "<name-for-your-service-principal>",
+    "appId": "<a-app-id>",
+    "appOwnerTenantId": "<a-tenant-id>",
+    "appRoleAssignmentRequired": false,
+    "appRoles": [],
+    "applicationTemplateId": null,
+    "deletionTimestamp": null,
+    "displayName": "<name-for-your-service-principal>",
+    "errorUrl": null,
+    "homepage": "https://<name-for-your-service-principal>",
+    "informationalUrls": {
+      "marketing": null,
+      "privacy": null,
+      "support": null,
+      "termsOfService": null
+    },
+    "keyCredentials": [],
+    "logoutUrl": null,
+    "notificationEmailAddresses": [],
+    "oauth2Permissions": [
+      {
+        "adminConsentDescription": "Allow the application to access <name-for-your-service-principal> on behalf of the signed-in user.",
+        "adminConsentDisplayName": "Access <name-for-your-service-principal>",
+        "id": "<an-id>",
+        "isEnabled": true,
+        "type": "User",
+        "userConsentDescription": "Allow the application to access <name-for-your-service-principal> on your behalf.",
+        "userConsentDisplayName": "Access <name-for-your-service-principal>",
+        "value": "user_impersonation"
+      }
+    ],
+    "objectId": "<a-object-id>",
+    "objectType": "ServicePrincipal",
+    "odata.type": "Microsoft.DirectoryServices.ServicePrincipal",
+    "passwordCredentials": [],
+    "preferredSingleSignOnMode": null,
+    "preferredTokenSigningKeyEndDateTime": null,
+    "preferredTokenSigningKeyThumbprint": null,
+    "publisherName": "<a-publisher-name>",
+    "replyUrls": [],
+    "samlMetadataUrl": null,
+    "samlSingleSignOnSettings": null,
+    "servicePrincipalNames": [
+      "http://<name-for-your-service-principal>",
+      "<a-service-principal-id>"
+    ],
+    "servicePrincipalType": "Application",
+    "signInAudience": "AzureADMyOrg",
+    "tags": [],
+    "tokenEncryptionKeyId": null
+  }
+]
+```
+
+From the resultant JSON output to the screen, copy the value of `objectID`.
+
+You can then run:
+
+```bash
+$ az role assignment create --assignee "<objectId>" --role "Storage Blob Data Contributor"
+$ az role assignment create --assignee "<objectId>" --role "User Access Administrator"
+$ az role assignment create --assignee "<objectId>" --role  "Azure Kubernetes Service Cluster User Role"
+```
+
+You should now be setup with the correct permissions to deploy the infrastructure onto Azure. Further reading on Azure Service Principals can be found [here](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest).
+
+### `deployment.tfvars.json` file
```

Comment:
Is this playing the same role as https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret? If so, I have a slight preference to use the environment there for authenticating the service principal.

Reply:
So the only real reason I'm passing it in as a variable is that https://learn.hashicorp.com/tutorials/terraform/aks#set-up-and-initialize-your-terraform-workspace the K8s cluster takes it as a parameter... The `az login` handles the actual authenticating for terraform to deploy on your behalf afaik. You just need to pass it to the cluster.

If I've understood correctly.

Reply:
Ah I misunderstood, this is for AKS.

I've only ever used "SystemAssigned" / managed identity: 


```hcl
  identity {
    type = "SystemAssigned"
  }
```

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster

https://docs.microsoft.com/en-us/azure/aks/use-managed-identity

That seemed a bit simpler, and I don't think I've had any issues.

Reply:
Ahh. Okay. I'll remove the AppId etc then from the vars file.

Reply:
Done now @TomAugspurger cheers